src/java.base/share/classes/java/util/stream/Gatherer.java

```diff
@@ -0,0 +1,566 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.util.stream;
+
+import jdk.internal.javac.PreviewFeature;
+import jdk.internal.vm.annotation.ForceInline;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Supplier;
+
+
+/**
+ * An intermediate operation that processes input elements, optionally mutating
+ * intermediate state, optionally transforming the input elements into a
+ * different type of output elements, and optionally applies final actions at
+ * end-of-upstream. Gatherer operations can be performed either sequentially,
+ * or be parallelized -- if a combiner function is supplied.
+ *
+ * <p>Examples of gathering operations include, but is not limited to:
+ * grouping elements into batches, also known as windowing functions;
+ * de-duplicating consecutively similar elements; incremental accumulation
+ * functions; incremental reordering functions, etc.  The class
+ * {@link java.util.stream.Gatherers} provides implementations of common
+ * gathering operations.
+ *
+ * @apiNote
+ * <p>A {@code Gatherer} is specified by four functions that work together to
+ * process input elements, optionally using intermediate state, and optionally
+ * perform a final operation at the end of input.  They are: <ul>
+ *     <li>creation of a new, potentially mutable, state ({@link #initializer()})</li>
+ *     <li>integrating a new input element ({@link #integrator()})</li>
+ *     <li>combining two states into one ({@link #combiner()})</li>
+ *     <li>performing an optional final operation ({@link #finisher()})</li>
+ * </ul>
```

Comment:
consistency: either all should be like "creation of a", or all should be an "ing" term (integrating, combining, performing)"